oneanddone/tasks/models.py

```diff
@@ -312,11 +311,40 @@ def __unicode__(self):
 
 
 class TaskTeam(CreatedModifiedModel, CreatedByModel):
+    description = models.TextField(blank=True)
     name = models.CharField(max_length=255)
+    url_code = models.CharField(max_length=30, verbose_name='team url suffix',
+                                unique=True, null=True)
 
     def __unicode__(self):
         return self.name
 
+    @property
+    def description_html(self):
+        return self._yield_html(self.description)
+
+    def get_absolute_url(self):
+        return reverse('tasks.team', args=[self.id])
+
+    def get_edit_url(self):
+        return reverse('tasks.edit_team', args=[self.id])
+
+    def get_friendly_url(self):
+        return reverse('tasks.team_short', args=[self.url_code])
+
+    @classmethod
+    def get_team_by_id_or_url_code(self, kwargs):
+        url_code = kwargs.get('url_code')
+        if url_code is None:
+            try:
+                return TaskTeam.objects.get(id=kwargs.get('pk'))
+            except ObjectDoesNotExist:
```

Comment:
I've been bitten in the past while using the generic ObjectDoesNotExist exception and catching one that was raised for another object than the one I intended it to. This of course is a very simple try/except nevertheless I'd still go with TaskTeam.DoesNotExist